### PR TITLE
Geoff Adebonojo Morse Translator

### PR DESCRIPTION
**Pull request by geoffreyadebonojo-zz**

### CODE REVIEW

File: lib/morse_translator.rb [hunk 1]
```diff
@@ -38,6 +38,41 @@ def initialize
...
+
+
+  def compare_units(element, letter)
+    if element == letter
+      @morse_string += @dictionary[letter]
+    end
+  end
+
+
+  def process_units(array_in)
```
Comment (memcmahon) [inline lib/morse_translator.rb:53]: This method seems to be doing quite a bit!  I start to sense some code smell anytime I see .each used multiple times in a single method!

File: lib/morse_translator.rb [hunk 2]
```diff
@@ -38,6 +38,41 @@ def initialize
...
+    letters = ('a'..'z').to_a
+    [1,2,3,4,5,6,7,8,9,0].to_a.each do |number|
```
Comment (memcmahon) [inline lib/morse_translator.rb:55]: Here, you are calling #to_a on an array - I think this is a bit unnecessary.  It also seems like a lot of this method is recreating a 'dictionary' type object which we give you above.

File: lib/morse_translator.rb [hunk 3]
```diff
@@ -38,6 +38,41 @@ def initialize
...
+    letters << " "
+    array_in.each do |element|
+      letters.each do |letter|
+        compare_units(element,letter)
+      end
+    end
+    @morse_string
+  end
+
+  def translate(string_in)
```
Comment (memcmahon) [inline lib/morse_translator.rb:67]: I like that you have turned this into a CEO method - it really shows me that you understand return values and how methods can interact with each other!

File: lib/morse_translator.rb [hunk 4]
```diff
@@ -38,6 +38,41 @@ def initialize
                     "9" => "----.",
                     "0" => "-----",
                     " " => " "}
+    @morse_string = ""
```
Comment (memcmahon) [inline lib/morse_translator.rb:41]: I'm not sure about having this as an instance variable - it seems like it should just be the return value of the #translate method.

File: test/morse_translator_test.rb [hunk 5]
```diff
@@ -0,0 +1,30 @@
+require 'pry'
+
+
+require 'minitest/autorun'
+require 'minitest/pride'
+
+require './lib/morse_translator'
+
+class MorseTranslatorTest < Minitest::Test
```
Comment (memcmahon) [inline test/morse_translator_test.rb:9]: You created 3 helper methods in your translator - I would have expected to see tests for each of those methods here!

**Review by memcmahon — COMMENTED** [review]
Geoff - I think you may have overcomplicated this problem!  You could accomplish the same thing with a much simplified translate method:

```ruby
def translate(english_string)
  letter_array = english_string.chars
  letter_array.map do |letter|
    @dictionary[letter]
  end.join
end
```

**Comment by geoffreyadebonojo-zz** [comment]
Oh man. Yeah that times assessment was a throat punch. Fun though. Eye
opening. Thanks for the feedback!

**Pull request closed without merging**